src/range.cr

```diff
@@ -258,7 +257,7 @@ struct Range(B, E)
       raise ArgumentError.new("Can't step beginless range")
     end
 
-    if start.responds_to?(:step)
+    if start.is_a?(Steppable)
```

Comment:
💭 @straight-shoota Because this check happens at runtime, the return value is a union, so the iterator is slower than not using a union. Is there a way to move this check to compile-time in some cases? For example by checking if `B < Steppable`. What do you think?

Reply:
That sounds like a very good idea. There are already similar cases in other ˋRangeˋ methods.